core/src/main/scala/org/apache/spark/storage/DiskStore.scala

```diff
@@ -71,6 +74,13 @@ private[spark] class DiskStore(
     logDebug(s"Attempting to put block $blockId")
     val startTimeNs = System.nanoTime()
     val file = diskManager.getFile(blockId)
+
+    // SPARK-37618: If fetching cached RDDs from the shuffle service is enabled, we must
+    // make the file world readable, as it will not be owned by the gropu running the shuffle
+    // service in a secure environment
```

Comment:
nit perhaps expand to say something like: due to changing directory permissions to allow deletion

Reply:
added, let me know how it sounds